wordpress-mobile/WordPress-iOS
# UL&S: Remove .showMagicLinkFlow segue

Fixes # n/a

This PR removes the segue `.showMagicLinkFlow`, which sends the user to the `LoginLinkViewController`. Now the user is navigated programmatically.

Ref. https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/256

### To test:

1. Check out this branch
2. `rake dependencies`
3. Build and run
4. Log out if you're logged in
5. Log in > Continue with WordPress.com > enter an email address > Next
6. **Expected:** the Magic Link screen will appear
7. Send Link > open the magic link (note: if 2fa is active, the magic link will ask for a 2fa code first. If the code is correct, then it will redirect you and reopen the WordPress app).
8. **Expected:** you are successfully navigated to the site list / "logged in as.." screen.

### Do not merge until:
2. The Authenticator PR has been merged: Ref: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/256
3. A new Authenticator release has been created 
4. The release is published on Cocoapods trunk 
5. This PR points to the new release  

PR submission checklist:
